HW_17/conftest.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import random
+import string
+from utilities.config_reader import get_application_url, get_browser_id, get_user_creds
+from utilities.driver_factory import driver_factory
+from page_objects.login_page import LoginPage
+from page_objects.main_page import MainPage
+from page_objects.category_page import CategoryPage
+from page_objects.account_page import AccountPage
+from page_objects.wishlist_page import WishlistPage
+
+
+@pytest.fixture()
+def create_browser():
+    driver = driver_factory(get_browser_id())
+    driver.maximize_window()
+    driver.get(get_application_url())
+    yield driver
+    driver.quit()
+
+
+@pytest.fixture()
+def generate_random_email():
+    random_email = [''.join(random.choice(string.ascii_letters) for _ in range(10))]
+    random_email += "@fake.com"
+    return random_email
+
+
+@pytest.fixture()
+def open_login_page(create_browser):
+    return LoginPage(create_browser)
+
+
+@pytest.fixture()
+def expand_dropdown_menu(open_login_page):
+    return open_login_page.expand_dropdown_menu()
+
+
+@pytest.fixture()
+def open_main_page(open_login_page, expand_dropdown_menu):
+    return expand_dropdown_menu.login(get_user_creds()[0], get_user_creds()[1])
+
+
+@pytest.fixture()
+def open_wishlist_page(open_main_page, create_browser):
+    return MainPage(create_browser).click_wishlist_button()
```

Comment:
you have open_main_page fixture here, what does it return? 
I guess it should return MainPage object
So you can return open_main_page.click_wishlist_button()

Please take a look to these all around the conftest